ui_testing/testcases/basic_tests/test05_testplans.py

```diff
@@ -51,8 +51,417 @@ def test001_test_plan_delete_testunit(self):
+        self.test_plan.sleep_small()
+
+        self.test_plan.archive_selected_items()
+        self.test_plan.get_archived_items()
+
+        archived_row = self.test_plan.search(testplan_number)
+        self.test_plan.sleep_small()
+        self.base_selenium.LOGGER.info('Checking if testplan number: {} is archived correctly'.format(testplan_number))
+        self.assertIsNotNone(archived_row[0])
+        self.base_selenium.LOGGER.info('Testplan number: {} is archived correctly'.format(testplan_number))
+
+        # restore and navigate to active table
+        self.base_selenium.LOGGER.info('Selecting the row')
+        self.test_plan.click_check_box(source=archived_row[0])
+        self.test_plan.sleep_small()
+
+        self.test_plan.restore_selected_items()
+        self.test_plan.get_active_items()
+
+        restored_row = self.test_plan.search(testplan_number)
+        self.base_selenium.LOGGER.info('Checking if testplan number: {} is restored correctly'.format(testplan_number))
+        self.assertIsNotNone(restored_row[0])
+        self.base_selenium.LOGGER.info('Testplan number: {} is restored correctly'.format(testplan_number))
+
+    def test005_archive_restore_test_plan_multiple_records(self):
+        '''
+        LIMS-3506 Case 2
+        Archive and restore multiple records
+        '''
+        self.base_selenium.LOGGER.info('Choosing random multiple testplans table rows')
+        rows = self.test_plan.select_random_multiple_table_rows(element='test_plans:test_plans_table')
+        testplan_rows = rows[0]
+        testplans_numbers = []
+        for row in testplan_rows:
+            testplans_numbers.append(row['Test Plan No.'])
+        self.test_plan.sleep_small()
+
+        # archive and navigate to archived table
+        self.base_selenium.LOGGER.info('Testplan numbers: {} will be archived'.format(testplans_numbers))
+
+        self.test_plan.archive_selected_items()
+        self.test_plan.get_archived_items()
+
+        self.base_selenium.LOGGER.info(
+            'Checking if testplan numbers: {} are archived correctly'.format(testplans_numbers))
+
+        archived_rows = self.test_plan.search_for_multiple_rows(testplans_numbers, 1)
+
+        self.assertIsNotNone(archived_rows)
+        self.assertEqual(len(archived_rows), len(testplans_numbers))
+
+        self.test_plan.sleep_small()
+
+        self.base_selenium.LOGGER.info('Testplan numbers: {} are archived correctly'.format(testplans_numbers))
+
+        # restore and navigate to active table
+        self.test_plan.restore_selected_items()
+        self.test_plan.get_active_items()
+
+        self.base_selenium.LOGGER.info(
+            'Checking if testplan numbers: {} are restored correctly'.format(testplans_numbers))
+
+        restored_rows = self.test_plan.search_for_multiple_rows(testplans_numbers)
+        self.assertIsNotNone(restored_rows)
+        self.assertEqual(len(restored_rows), len(testplans_numbers))
```

Comment:
assert that the test plans numbers are existing

Reply:
Same as previous comment

Reply:
I'll use filter instead of search